plugins/create-plugin.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2023 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=invalid-name,missing-docstring,consider-using-f-string
+
+import os
+import glob
+
+subst = {}
+
+
+def subst_add_string(key: str, value: str) -> None:
+
+    subst[key] = value
+    subst[key.lower()] = value.lower()
+    subst[key.upper()] = value.upper()
+
+
+def subst_replace(data: str) -> str:
+
+    for key, value in subst.items():
+        data = data.replace(key, value)
+    return data
+
+
+subst_add_string("Xvendor", "Hughski")
+subst_add_string("Xbasename", "Colorhug")
+subst_add_string("Xparent", "Usb")
+subst_add_string("Xyear", "2023")
+subst_add_string("Xauthor", "Richard Hughes")
+subst_add_string("Xemail", "richard@hughsie.com")
```

Comment:
I think we're going to want something with `argparse` args for all this stuff that gets populated